### PR TITLE
EntityInstantiator

### PR DESCRIPTION
**Pull request by KYPremco**
Changed name from Modifier to instantiator because it doesn't have to modify. And it hooks into the Instantiotor of Entity service hence EntityInstantiator.

Made the EntityInstantiator class internal to prevent confusion since it doesn't need to be used for the users.

### CODE REVIEW

File: TimberAPI/EntityInstantiator/EntityInstantiator.cs [hunk 1]
```diff
@@ -0,0 +1,31 @@
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using HarmonyLib;
+using Timberborn.EntitySystem;
+using UnityEngine;
+
+namespace TimberbornAPI.EntityInstantiatorSystem
+{
+    internal class EntityInstantiator
```
Comment (thundersen) [inline TimberAPI/EntityInstantiator/EntityInstantiator.cs:10]: For me an `EntityInstantiator` would be a class that creates entity instances. But this class applies actions to `GameObject`s *when* they're instantiated.
How about `GameObjectActionApplier` for this class? Or something more catchy like `GameObjectTweaker`?
And consequently:
`TimberbornAPI.EntityInstantiatorSystem` -> `TimberbornAPI.GameObjectActions` or `TimberbornAPI.GameObjectTweaking`?
`IEntityInstantiator` -> `IGameObjectAction`?
`_gameObjectModifiers` -> `_instantiationActions`?
`visualElementInitializers` -> `instantiationActions`?

File: TimberAPIExample/Examples/GameObjectModifier/EntityInstantiatorConfigurator.cs [hunk 2]
```diff
@@ -0,0 +1,15 @@
+using Bindito.Core;
+using TimberAPIExample.AutoConfiguratorInstaller;
+using TimberbornAPI.EntityInstantiatorSystem;
+
+namespace TimberAPIExample.Examples.GameObjectModifier
+{
+    public class EntityInstantiatorConfigurator : IInGameConfigurator
```
Comment (thundersen) [inline TimberAPIExample/Examples/GameObjectModifier/EntityInstantiatorConfigurator.cs:7]: Don't forget to rename this as well, if we agree on something else than `EntityInstantiator`

**Review by thundersen — CHANGES_REQUESTED** [review]
Excellent stuff. Especially the examples are really great for understanding what this can do.
But I would suggest some naming changes.

**Comment by KYPremco** [comment]
- Moved namspaces to folders
- Renamed the actions classes, interfaces
- Fixed creative construction example in update

**Review by thundersen — APPROVED** [review]

**Pull request merged**